src/core/pdf/file_uploader.py

```diff
@@ -0,0 +1,29 @@
+from attrs import define
+import streamlit as st
+from src.states.gui import GUIState
+
+
+@define
+class FileUploader:
+    state: GUIState
+
+    def __attrs_post_init__(self):
+        uploader_state = self.state.left_col.uploader
+        chat_state = self.state.left_col.chat
+        rerun = self.state.rerun
+
+        uploaded_file = st.file_uploader(
```

Comment:
Let's consider the case when the file was replaced with another one. In this case I believe we also should reset the chat.